app/src/main/java/org/wikipedia/edit/insertmedia/InsertMediaViewModel.kt

```diff
@@ -63,20 +60,22 @@ class InsertMediaViewModel(bundle: Bundle) : ViewModel() {
                         }
                         pageTitle
                     }
-                    LoadResult.Page(results, null, response.continuation)
+                    LoadResult.Page(results, params.key, if (response.continuation?.gsroffset != 0)
+                        response.continuation?.gsroffset else null)
```

Comment:
Not sure but would it be the same if we pass `response.continuation?.gsroffset` directly and it will output either `0` or `null` initially?

Reply:
Yep. This necessitated a few updates/improvements in the main Search logic, as well.